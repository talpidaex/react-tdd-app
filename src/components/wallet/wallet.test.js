import { renderWithProviders } from "../../redux/utils/test-util";
import Wallet from "./wallet";
import { screen } from "@testing-library/react";

describe("Wallet", () => {
  const initialState = { balance: 0 };

  function Component() {
    return renderWithProviders(<Wallet />, { initialState });
  }

  it("renders properly", () => {
    const { container } = Component();
    expect(container).toMatchSnapshot();
  });

  it("displays the balance from useSelector", () => {
    Component();
    expect(screen.getByTestId("balance-text").innerHTML).toEqual("Wallet balance:0");
  });
});
