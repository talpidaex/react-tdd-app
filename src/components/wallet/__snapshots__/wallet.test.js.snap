// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders properly 1`] = `
<div>
  <div
    data-testid="wallet-component"
  >
    <h3
      data-testid="balance-text"
    >
      Wallet balance:
      0
    </h3>
  </div>
</div>
`;
